gate-oauth2/src/main/groovy/com/netflix/spinnaker/gate/security/oauth2/SpinnakerUserInfoTokenServices.groovy

```diff
@@ -1,264 +0,0 @@
-      if (providerTokenServices != null && !providerTokenServices.hasAllProviderRequirements(accessToken, details)) {
-        throw new BadCredentialsException("User's provider info does not have all required fields.")
-      }
-    }
-
-    def username = details[userInfoMapping.username] as String
-    def roles = Optional.ofNullable(groupExtractor)
-      .map({ extractor -> extractor.apply(accessToken, details) })
-      .or({ Optional.ofNullable(getRoles(details)) })
-      .orElse([])
-
-    // Service accounts are already logged in.
-    if (!isServiceAccount) {
-      def id = registry
-        .createId("fiat.login")
-        .withTag("type", "oauth2")
-
-      try {
-        retrySupport.retry({ ->
-          if (roles.isEmpty()) {
-            permissionService.login(username)
-          } else {
-            permissionService.loginWithRoles(username, roles)
-          }
-        }, 5, 2000, false)
-        log.debug("Successful oauth2 authentication (user: {}, roleCount: {}, roles: {})", username, roles.size(), roles)
-        id = id.withTag("success", true).withTag("fallback", "none")
-      } catch (Exception e) {
-        log.debug(
-          "Unsuccessful oauth2 authentication (user: {}, roleCount: {}, roles: {}, legacyFallback: {})",
-          username,
-          roles.size(),
-          roles,
-          fiatClientConfigurationProperties.legacyFallback,
-          e
-        )
-        id = id.withTag("success", false).withTag("fallback", fiatClientConfigurationProperties.legacyFallback)
-
-        if (!fiatClientConfigurationProperties.legacyFallback) {
-          throw e
-        }
-      } finally {
-        registry.counter(id).increment()
-      }
-    }
-
-    User spinnakerUser = new User(
-        email: details[userInfoMapping.email] as String,
-        firstName: details[userInfoMapping.firstName] as String,
-        lastName: details[userInfoMapping.lastName] as String,
-        allowedAccounts: allowedAccountsSupport.filterAllowedAccounts(username, roles),
-        roles: roles,
-        username: username)
-
-    PreAuthenticatedAuthenticationToken authentication = new PreAuthenticatedAuthenticationToken(
-        spinnakerUser,
-        null /* credentials */,
-        spinnakerUser.authorities
-    )
-
-    // impl copied from UserInfoTokenServices
-    OAuth2Request storedRequest = new OAuth2Request(null, sso.clientId, null, true /*approved*/,
-                                                    null, null, null, null, null);
-
-    return new OAuth2Authentication(storedRequest, authentication)
-  }
-
-  @Override
-  OAuth2AccessToken readAccessToken(String accessToken) {
-    return userInfoTokenServices.readAccessToken(accessToken)
-  }
-
-  boolean isServiceAccount(Map details) {
-    String email = details[userInfoMapping.serviceAccountEmail]
-    if (!email || !permissionService.isEnabled()) {
-      return false
-    }
-    try {
-      def serviceAccounts = front50Service.getServiceAccounts()
```

Comment:
Interesting.  I would have figured https://github.com/spinnaker/gate/pull/1866 changed this to
```
Retrofit2SyncCall.execute(front50Service.getServiceAccounts())
```
but apparently it didn't... cc: @kirangodishala

Reply:
https://github.com/spinnaker/gate/pull/1873 took care of the change.